MvvmCross/Core/Core/ViewModels/MvxViewModel.cs

```diff
@@ -82,10 +83,62 @@ public abstract class MvxViewModel<TInit> : MvxViewModel, IMvxViewModelInitializ
                 throw new MvxIoCResolveException("There is no implementation of IMvxJsonConverter registered. You need to use the MvvmCross Json plugin or create your own implementation of IMvxJsonConverter.");
             }
 
-            var deserialized = serializer.DeserializeObject<TInit>(parameter);
-            await Init(deserialized);
+            var deserialized = serializer.DeserializeObject<TParameter>(parameter);
+            await Initialize(deserialized);
+        }
+
+        public abstract Task Initialize(TParameter parameter);
+    }
+
+    //TODO: Not possible to name MvxViewModel, name is MvxViewModelResult for now
+    public abstract class MvxViewModelResult<TResult> : MvxViewModel, IMvxViewModelResult<TResult> where TResult : class
```

Comment:
IMvxViewModelResult implies to me that it **is a result**. Not sure about the best name, but even IMvxViewResultViewModel already feels better.
Also, regarding the TODO, I would actually prefer a name that clearly indicates that this is a viewmodel that is expected to return a value on Close.